File: Algorithms/701. Insert into a Binary Search Tree/golang/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func insertIntoBST(root *TreeNode, val int) *TreeNode {
	node := root
	for node != nil {
		if val > node.Val {
			if node.Right == nil {
				node.Right = &TreeNode{Val: val}
				return root
			} else {
				node = node.Right
			}
		} else {
			if node.Left == nil {
				node.Left = &TreeNode{Val: val}
				return root
			} else {
				node = node.Left
			}
		}
	}

	return &TreeNode{Val: val}
}

File: Algorithms/701. Insert into a Binary Search Tree/javascript/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function (root, val) {
    let node = root;
    while (node) {
        if (val > node.val) {
            if (node.right) {
                node = node.right;
            } else {
                node.right = new TreeNode(val);
                return root;
            }
        } else {
            if (node.left) {
                node = node.left;
            } else {
                node.left = new TreeNode(val);
                return root;
            }
        }
    }
    return new TreeNode(val);
};
